### PR TITLE
MAT-4292 system module setup.

### PR DESCRIPTION
**Pull request by mcmcphillips**
## MADiE PR

Jira Ticket: [MAT-4292](https://jira.cms.gov/browse/MAT-4292)
(Optional) Related Tickets:

[madie-layout](https://github.com/MeasureAuthoringTool/madie-layout/pull/106)
[madie-measure](https://github.com/MeasureAuthoringTool/madie-measure/pull/97)
[madie-root](https://github.com/MeasureAuthoringTool/madie-root/pull/82)

### Summary
* sets up a basic store to track measure context outside of measures so pageheader can be aware of content to display.
* adds type declarations
* consolidates hooks and config for measures and layout to use

May need to run tsc:build before it becomes self aware.

### CODE REVIEW

**Comment by jkotanchik-SB** [comment]
@mcmcphillips Is `madie-madie-frontend-template.tsx` still used?

**Comment by jkotanchik-SB** [comment]
@mcmcphillips There is another ServiceConfig in `madie-cql-library` that exposes a `cqlLibraryService` config.

File: src/Config/Config.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+export interface ServiceConfig {
+  measureService: {
+    baseUrl: string;
+  };
+  elmTranslationService: {
+    baseUrl: string;
+  };
+}
```
Comment (jkotanchik-SB) [inline src/Config/Config.ts:20]: There is another ServiceConfig in `madie-cql-library` that exposes a `cqlLibraryService` config, though it is set up as a Context.

**Review by jkotanchik-SB — DISMISSED** [review]

File: src/Config/Config.ts (hunk 1 above)
Comment (jkotanchik-SB) [reply src/Config/Config.ts:20]: Found another config entry:  
```
 terminologyService: {
    baseUrl: string;
  };
```

Comment (mcmcphillips) [reply src/Config/Config.ts:20]: Added accordingly.
We will want to retrieve config from util in a later ticket

**Review by adongare — APPROVED** [review]

**Pull request merged**